app/scripts/controllers/groups/AddRoleController.js

```diff
@@ -2,16 +2,37 @@
     mifosX.controllers = _.extend(module, {
         AddRoleController: function (scope, routeParams, location, resourceFactory) {
             scope.formData = {};
-            resourceFactory.groupResource.get({groupId: routeParams.id, associations: 'all', template: 'true'}, function (data) {
+            resourceFactory.groupResource.get({
+                groupId: routeParams.id,
+                associations: 'all',
+                template: 'true'
+            }, function (data) {
                 scope.group = data;
                 scope.clients = data.clientMembers;
                 scope.roles = data.availableRoles;
-                scope.formData.clientId = data.clientMembers[0].id;
-                scope.formData.role = data.availableRoles[0].id;
+
+                /**
+                 * Checking availableRoles[0] is null before applying it to the main scope.
+                 * Otherwise there's a
+                 */
+                if (data.availableRoles != "") {
+                    scope.formData.role = data.availableRoles[0].id;
```

Comment:
if data.availableRoles = [] then this change will throw error as well... data.availableRoles && data.availableRoles.length > 0 could be a better check